### PR TITLE
[Merged by Bors] - config: rename `--test-mode` flag

### LINKED ISSUE #3026: Rename `test-mode` flag
## Description

Rename the `test-mode` flag that makes logger output all logs as JSON's to something more self-descriptive. Not urgent. We need to ensure that this action doesn't break anything in the infrastructure

https://github.com/spacemeshos/go-spacemesh/issues/3024#issuecomment-1001620954

## Affected code

Node args handling, configs, infrastructure

This issue appears in commit hash: 
`9972b2c9c7ee79a2ef913eeb0d8ead05ee74769c`

### PR DESCRIPTION
**Pull request by nkryuchkov**
## Motivation

Closes #3026

The `--test-mode` flag just makes logger output logs as JSON instead of plaintext, so its name is not clear. This PR renames it to `--json-log`.

## Changes

- Rename `--test-mode` flag to `--json-log`

## Test Plan

Unit tests, system tests, all Github actions.

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by dshulyak** [comment]
We have `logging` section in config. I think it is better to reuse it and enable json output with logging.json=true or logging.log(-encoder)=json | console, whichever makes more sense

**Comment by nkryuchkov** [comment]
@dshulyak something like https://github.com/spacemeshos/go-spacemesh/pull/3056/commits/43335bb32d193bd4b2b776f33417e0498b2e4eaa?

File: config/logging.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package config
+
+import "go.uber.org/zap/zapcore"
+
+// LogEncoder defines a log encoder kind.
+type LogEncoder = string
+
+const (
+	// PlainTextLogEncoder represents logging with plain text.
+	PlainTextLogEncoder LogEncoder = "plain"
```
Comment (dshulyak) [inline config/logging.go]: it is called console encoder in zap

**Review by dshulyak — APPROVED** [review]

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**